### PR TITLE
Checkboxes: update checkbox design and remove noticons

### PR DESCRIPTION
**Pull request by drw158**
This removes noticons from checkboxes. Before, the icon was being injected with the noticon font in the CSS. The easiest way to remove noticons is to replace `content` with an svg file.

```content: url( "/calypso/images/checkbox-icons/checkmark.svg" );```

This is not ideal, but currently the best solution in lieu of creating a new checkbox component. Hopefully this isn't permanent and we can implement that later.

I don't imagine the `checkmark` icon will be updated anytime soon, so I think it'll be ok for now.

Before
bulkselects

After
bulkselects-new

### CODE REVIEW

File: assets/stylesheets/shared/_forms.scss [hunk 1]
```diff
@@ -159,19 +159,23 @@ input[type=radio] + span {
...
-		content: '\f418';
-		margin: -4px 0 0 -5px;
-		float: left;
+	&:checked {
+		border-color: $blue-medium;
+		background-color: $blue-medium;
+	}
+
+	&:checked:before {
+		content: url( "/calypso/images/checkbox-icons/checkmark.svg" );
```
Comment (shaunandrews) [inline assets/stylesheets/shared/_forms.scss]: Is there a better way to do this without injecting the SVG in with CSS? Is there something stopping us from hiding/rendering the checkbox in the jsx?

Comment (shaunandrews) [reply assets/stylesheets/shared/_forms.scss]: My biggest concern is that we're duplicating a Gridicon in the `/images/` folder that will need to be manually updated if Gridicon's changes.

File: client/components/bulk-select/index.jsx [hunk 2]
```diff
@@ -54,7 +54,6 @@ export default class extends React.Component {
 				<span className="bulk-select__container">
 					<input type="checkbox" className={ inputClasses } checked={ isChecked } readOnly />
 					<Count count={ this.props.selectedElements } />
-					{ this.getStateIcon() }
```
Comment (scruffian) [inline client/components/bulk-select/index.jsx]: If we are going to remove this we should remove the `getStateIcon` function too.

**Comment by scruffian** [comment]
I decided to simplify this so that we can just remove the Noticons and iterate later. I appreciate that this means duplicating the Gridicon SVG, which is a compromise but I think its worth it to get rid of Noticons:

Before
bulkselects

After
bulkselects-new

**Comment by shaunandrews** [comment]
> I think its worth it to get rid of Noticons

Agreed.

**Comment by shaunandrews** [comment]
I noticed we added a border-radius to the checkboxes. I don't mind it, but wonder if there was a rationale behind adding it?

**Comment by drw158** [comment]
Design LGTM. I tested /comments and /plugins.

Thanks for working on this! Your approach is much simpler than mine.

**Pull request merged**